core/unit_test/TestViewIsAssignable.hpp

```diff
@@ -94,6 +94,14 @@ TEST(TEST_CATEGORY, view_is_assignable) {
   // Layout assignment
   Impl::TestAssignability<View<int*, left, d_exec>,
                           View<int*, right, d_exec>>::test(true, true, 10);
+  Impl::TestAssignability<View<int[5], left, d_exec>,
+                          View<int*, right, d_exec>>::test(false, false, 10);
+  Impl::TestAssignability<View<int[10], left, d_exec>,
+                          View<int*, right, d_exec>>::test(false, true, 10);
+  Impl::TestAssignability<View<int*, left, d_exec>,
+                          View<int[5], right, d_exec>>::test(true, true, 5);
```

Comment:
Wait why do you pass `5`?  Isn't it something we only relaxed not so long ago?  https://github.com/kokkos/kokkos-core-wiki/pull/108
Please drop it here and below.

Reply:
yeah unnescessary here, gonna change that.

Reply:
Also added the rank-0 mixed layout case.